tools/ports/icu.py

```diff
@@ -20,24 +24,74 @@ def get(ports, settings, shared):
   url = 'https://github.com/unicode-org/icu/releases/download/%s/icu4c-%s-src.zip' % (TAG, VERSION)
   ports.fetch_project('icu', url, 'icu', sha512hash=HASH)
 
-  def create(final):
-    logging.info('building port: icu')
+  def get_downloaded_icu4c_path():
+    return os.path.join(ports.get_dir(), 'icu', 'icu')
+
+  def get_icu_build_path():
+    return os.path.join(ports.get_build_dir(), 'icu')
 
-    source_path = os.path.join(ports.get_dir(), 'icu', 'icu')
-    dest_path = os.path.join(ports.get_build_dir(), 'icu')
+  def get_icu_source_path():
+    return os.path.join(get_icu_build_path(), 'source')
```

Comment:
Could these three function not just be local variables?  e.g.:

```
downloaded_icu4c_path = os.path.join(ports.get_dir(), 'icu', 'icu')
```

It seems like making them function make things more verbose and less efficient (due to re-computation).